src/auth/googleauth.ts

```diff
@@ -365,9 +361,17 @@ export class GoogleAuth<T extends AuthClient = JSONClient> {
     // For GCE, just return a default ComputeClient. It will take care of
     // the rest.
     (options as ComputeOptions).scopes = this.getAnyScopes();
-    this.cachedCredential = new Compute(options);
-    projectId = await this.getProjectIdOptional();
-    return {projectId, credential: this.cachedCredential};
+    return this.prepareADCResponse(new Compute(options));
+  }
+
+  private async prepareADCResponse(
+    credential: JSONClient | Impersonated | Compute | T
+  ): Promise<ADCResponse> {
+    this.cachedCredential = credential;
+    const projectId = await this.getProjectIdOptional();
+    credential.setQuotaProjectIdFromEnvironment();
```

Comment:
For clarify, let's move this before `this.cachedCredential` assignment (even if it does not change the behavior).